app/src/main/java/mozilla/lockbox/adapter/ItemListAdapter.kt

```diff
@@ -92,14 +114,19 @@ class ItemListAdapter : RecyclerView.Adapter<ItemListCell>() {
 
     override fun getItemViewType(position: Int): Int {
         val count = itemList?.count() ?: 0
-        return when (count) {
-            0 -> if (isFiltering) NO_MATCHING_ENTRIES_CELL_TYPE else NO_ENTRIES_CELL_TYPE
-            else -> ITEM_DISPLAY_CELL_TYPE
+        if (count > 0) {
+            return ITEM_DISPLAY_CELL_TYPE
+        }
+
+        return when (type) {
+            is ItemListAdapterType.ItemList -> NO_ENTRIES_CELL_TYPE
+            is ItemListAdapterType.Filter -> NO_MATCHING_ENTRIES_CELL_TYPE
+            is ItemListAdapterType.AutofillFilter -> SIMPLE_NO_ENTRIES_CELL_TYPE
         }
     }
 
-    fun updateItems(newItems: List<ItemViewModel>, isFiltering: Boolean = false) {
-        this.isFiltering = isFiltering
+    fun updateItems(newItems: List<ItemViewModel>, type: ItemListAdapterType = ItemListAdapterType.ItemList) {
+        this.type = type
```

Comment:
How often does `this.type` change. I think I would _prefer_ this as an instance variable passed in to the adapter's constructor.